assignments/02-collections-lifetimes/src/life.rs

```diff
@@ -1,19 +1,98 @@
-fn split_string(string: &str, delimeter: &str) -> Vec<&str> {
-    todo!()
+#[allow(dead_code)]
+fn split_string<'a>(string: &'a str, delimeter: &str) -> Vec<&'a str> {
+    let mut output = Vec::<&str>::new();
+    let mut start_index = 0;
+    let mut end_index;
+
+    while start_index < string.len() {
+        if let Some(index) = string[start_index..].find(delimeter) {
+            end_index = start_index + index;
+            output.push(&string[start_index..end_index]);
+            start_index = end_index + delimeter.len();
+        } else {
+            end_index = string.len();
+            output.push(&string[start_index..end_index]);
+            break;
+        }
+    }
+
+    output
 }
 
 #[derive(PartialEq, Debug)]
-struct Differences {
-    only_in_first: Vec<&str>,
-    only_in_second: Vec<&str>,
+struct Differences<'a> {
+    only_in_first: Vec<&'a str>,
+    only_in_second: Vec<&'a str>,
```

Comment:
note - this works fine, but since these two don't technically need to live in the same lifetime we can also separate them into two lifetimes as so:

```
struct Differences<'a, 'b> {
    only_in_first: Vec<&'a str>,
    only_in_second: Vec<&'b str>,
}
```

(not counting off, just wanted to point this out bc it could be useful for future functions)